server/src/instrument/instrument.controller.ts

```diff
@@ -0,0 +1,60 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, Put, HttpException, HttpStatus, UseGuards, Request, UnauthorizedException} from '@nestjs/common';
+import { InstrumentService } from './instrument.service';
+import { CreateInstrumentDto } from './dto/create-instrument.dto';
+import { UpdateInstrumentDto } from './dto/update-instrument.dto';
+import { Roles } from '../authorization/roles.decorator';
+import { RolesGuard } from '../authorization/roles.guard';
+import { JwtAuthGuard } from '../authentication/jwt-auth.guard';
+import { Role } from 'src/authorization/role.enum';
+
+@Controller('instrument')
+export class InstrumentController {
+
+  constructor(private instrumentService: InstrumentService) {}
+
+  @Get()
+  @Roles(Role.Admin, Role.User)
+  @UseGuards(JwtAuthGuard, RolesGuard)
+  async getInstruments(@Request() req: any){
+    return this.instrumentService.getInstruments();
+}
+
+  @Get(':id')
+  @Roles(Role.Admin)
+  @UseGuards(JwtAuthGuard, RolesGuard)
+  async getInstrument(@Request() req: any, @Param('id') id: string) {
+    if (req.role != Role.Admin){
```

Comment:
When you put only role.admin it knows to give unauthorized for non admin so no need for this